lib/ui/advanced_settings_screen.dart

```diff
@@ -0,0 +1,207 @@
+                              ),
+                            ),
+                          ],
+                        ),
+                      ],
+                    ),
+                  ),
+                );
+              },
+              childCount: 1,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Future<void> _showAlbumGridSizePicker() async {
+    final List<Text> options = [];
+    options.add(
+      Text("2", style: Theme.of(context).textTheme.subtitle1),
+    );
+    options.add(
+      Text("3", style: Theme.of(context).textTheme.subtitle1),
+    );
+    options.add(
+      Text("4", style: Theme.of(context).textTheme.subtitle1),
+    );
+    return showCupertinoModalPopup(
+      context: context,
+      builder: (context) {
+        return Column(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: <Widget>[
+            Container(
+              decoration: BoxDecoration(
+                color: Theme.of(context).colorScheme.cupertinoPickerTopColor,
+                border: const Border(
+                  bottom: BorderSide(
+                    color: Color(0xff999999),
+                    width: 0.0,
+                  ),
+                ),
+              ),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: <Widget>[
+                  CupertinoButton(
+                    onPressed: () {
+                      Navigator.of(context).pop('cancel');
+                    },
+                    padding: const EdgeInsets.symmetric(
+                      horizontal: 8.0,
+                      vertical: 5.0,
+                    ),
+                    child: Text(
+                      'Cancel',
+                      style: Theme.of(context).textTheme.subtitle1,
+                    ),
+                  ),
+                  CupertinoButton(
+                    onPressed: () async {
+                      await _config.setAlbumGridSize(_chosenGridSize);
+                      Bus.instance.fire(
+                        ForceReloadHomeGalleryEvent("grid size changed"),
+                      );
+                      _albumGridSize = _chosenGridSize;
+                      setState(() {});
+                      Navigator.of(context).pop('');
+                    },
+                    padding: const EdgeInsets.symmetric(
+                      horizontal: 16.0,
+                      vertical: 2.0,
+                    ),
+                    child: Text(
+                      'Confirm',
+                      style: Theme.of(context).textTheme.subtitle1,
+                    ),
+                  )
+                ],
+              ),
+            ),
+            Container(
+              height: 220.0,
+              color: const Color(0xfff7f7f7),
+              child: CupertinoPicker(
+                backgroundColor:
+                    Theme.of(context).backgroundColor.withOpacity(0.95),
```

Comment:
at all places, can we please use enteText theme and ente Color scheme?

Reply:
Done. Have used `backgroundElevated2` instead of `cupertinoPickerTopColor` because they look close enough (and I don't think this widget is on Figma yet)